opensearch/catalog-opensearch-endpoint/src/main/java/org/codice/ddf/endpoints/ResourceUriTransformerCatalogFrameworkProxy.java

```diff
@@ -0,0 +1,199 @@
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getEnterpriseResource(request);
+    }
+
+    public Map<String, Set<String>> getEnterpriseResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getEnterpriseResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getLocalResource(ResourceRequest request) throws IOException,
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getLocalResource(request);
+    }
+
+    public Map<String, Set<String>> getLocalResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getLocalResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getResource(ResourceRequest request, String resourceSiteName)
+        throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getResource(request, resourceSiteName);
+    }
+
+    public Map<String, Set<String>> getResourceOptions(String metacardId, String sourceId)
+        throws ResourceNotFoundException {
+        return catalog.getResourceOptions(metacardId, sourceId);
+    }
+
+    public Set<String> getSourceIds() {
+        return catalog.getSourceIds();
+    }
+
+    public SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest)
+        throws SourceUnavailableException {
+        return catalog.getSourceInfo(sourceInfoRequest);
+    }
+
+    public QueryResponse query(QueryRequest query) throws UnsupportedQueryException,
+        SourceUnavailableException, FederationException {
+        return catalog.query(query);
+    }
+
+    public QueryResponse query(QueryRequest queryRequest, FederationStrategy strategy)
+        throws SourceUnavailableException, UnsupportedQueryException, FederationException {
+        return catalog.query(queryRequest, strategy);
+    }
+
+    public BinaryContent transform(Metacard metacard, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        return catalog.transform(metacard, transformerId, requestProperties);
+    }
+
+    /**
+     * Converts the resource URIs found in the {@link SourceResponse} {@link Metacard}s using the
+     * {@link ActionProvider} injected in the constructor before delegating the call to the proxied
+     * {@link CatalogFramework} object.
+     */
+    public BinaryContent transform(SourceResponse response, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        transform(response);
+        return catalog.transform(response, transformerId, requestProperties);
+    }
+
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException,
+        SourceUnavailableException {
+        return catalog.update(updateRequest);
+    }
+
+    private void transform(SourceResponse queryResponse) {
```

Comment:
I have been following this thread and wanted to ask a couple of questions and present my thoughts.

It seems the information that you want (the download URL) is exactly what the Action Framework is designed to do.  So shouldn't the transformers use the Action Framework as designed, and then set the values accordingly in their specific output format?  In the case of the XML Transformer - perhaps it makes sense to have elements related to actions.  It seems like the proxy approach and even the additional attribute approach are mixing metacard concepts and muddying things up, rather than having the clear separate between Metacard Attributes and Actions that can be done on a Metacard.

I would think you would want to make the XML Transformer (and any transformers) Action aware.   Likewise any clients using that response format should be aware of the format of the client.

I don't think the Resource URL (which is retrieved by the Resource Action), should have any special handling that other Actions (thumbnail URL, Metadata URL, etc.) don't have.

Just a thought :)